python/paddle/fluid/tests/unittests/dygraph_to_static/test_cast.py

```diff
@@ -0,0 +1,149 @@
+# Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import unittest
+import numpy as np
+import paddle.fluid as fluid
+from paddle.fluid.dygraph import declarative
+
+SEED = 2020
+np.random.seed(SEED)
+
+
+@declarative
+def test_bool_cast(x):
+    x = fluid.dygraph.to_variable(x)
+    x = bool(x)
+    return x
+
+
+@declarative
+def test_int_cast(x):
+    x = fluid.dygraph.to_variable(x)
+    x = int(x)
+    return x
+
+
+@declarative
+def test_float_cast(x):
+    x = fluid.dygraph.to_variable(x)
+    x = float(x)
+    return x
+
+
+@declarative
+def test_mix_cast(x):
+    x = fluid.dygraph.to_variable(x)
+    x = int(x)
+    x = float(x)
+    x = bool(x)
+    x = float(x)
+    return x
+
```

Comment:
Add a test case that `x` is not a variable?

Reply:
Done.